code/controllers/subsystems/fail2topic.dm

```diff
@@ -84,12 +82,15 @@ var/datum/controller/subsystem/fail2topic/SSfail2topic
 /datum/controller/subsystem/fail2topic/proc/BanFromFirewall(ip)
 	if (!enabled)
 		return
+	var/static/regex/R = regex(@"(\.0\.)|(\.0$)|(\l+|/|;|&|\||-|%)") // Anything that interacts with a shell should be parsed. Prevents subnet banning and possible injection vulnerabilities
+	if(length(ip) > 15 || length(findtext(ip, R)))
+		return FALSE
 
 	active_bans[ip] = world.time
 	fail_counts -= ip
 	rate_limiting -= ip
 
-	. = shell("netsh advfirewall firewall add rule name=\"[rule_name]\" dir=in interface=any action=block remoteip=[ip]")
+	. = shell("netsh advfirewall firewall add rule name=\"[config.fail2topic_rule_name]\" dir=in interface=any action=block remoteip=[ip]")
```

Comment:
This may not fix the issue. If you can edit the configuration by any means (verbs or SDQL) then this just moves the edit location.

Reply:
Right, I probably should have checked the config type. Someone should probably update the controller xd